django-stubs/utils/functional.pyi

```diff
@@ -83,17 +85,16 @@ class SimpleLazyObject(LazyObject):
 _PartitionMember = TypeVar("_PartitionMember")
 
 def partition(
-    predicate: Callable[[_PartitionMember], Union[int, bool]], values: List[_PartitionMember]
-) -> Tuple[List[_PartitionMember], List[_PartitionMember]]: ...
+    predicate: Callable[[_PartitionMember], int | bool], values: list[_PartitionMember]
+) -> tuple[list[_PartitionMember], list[_PartitionMember]]: ...
 
 _Get = TypeVar("_Get", covariant=True)
-_Self = TypeVar("_Self")
 
 class classproperty(Generic[_Get]):
-    fget: Optional[Callable[[_Self], _Get]] = ...
-    def __init__(self, method: Optional[Callable[[_Self], _Get]] = ...) -> None: ...
-    def __get__(self, instance: Optional[_Self], cls: Type[_Self] = ...) -> _Get: ...
-    def getter(self, method: Callable[[_Self], _Get]) -> classproperty[_Get]: ...
+    fget: Callable[[Self], _Get] | None
+    def __init__(self: Self, method: Callable[[Self], _Get] | None = ...) -> None: ...
+    def __get__(self: Self, instance: Self | None, cls: type[Self] = ...) -> _Get: ...
+    def getter(self: Self, method: Callable[[Self], _Get]) -> classproperty[_Get]: ...
```

Comment:
These `self: Self` annotations seem to cause regression #1285, I have submitted a PR to fix it in #1287.